packages/flutter/lib/src/material/input_border.dart

```diff
@@ -246,9 +246,45 @@ class UnderlineInputBorder extends InputBorder {
     TextDirection? textDirection,
   }) {
     if (borderRadius.bottomLeft != Radius.zero || borderRadius.bottomRight != Radius.zero) {
-      canvas.clipPath(getOuterPath(rect, textDirection: textDirection));
+      _drawMultipleWidth(canvas, borderRadius.toRRect(rect), borderSide);
+    } else {
+      canvas.drawLine(rect.bottomLeft, rect.bottomRight, borderSide.toPaint());
     }
-    canvas.drawLine(rect.bottomLeft, rect.bottomRight, borderSide.toPaint());
+  }
+
+  void _drawMultipleWidth(Canvas canvas, RRect borderRect, BorderSide bottom) {
```

Comment:
Nit: Could this be static?

Reply:
Yes (but why?).

But this is going to probably be replaced by BoxBorder.paintNonUniformPath. I just need to get that approved. Shouldn't take more than a few days.